store/backup.go

```diff
@@ -8,34 +8,273 @@ import (
+		HookURL: raw.HookURL,
+	}
+}
+
+// CreateBackup creates an instance of Backup
+func (s *Store) CreateBackup(ctx context.Context, create *api.BackupCreate) (*api.Backup, error) {
+	backupRaw, err := s.createBackupRaw(ctx, create)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create Backup with BackupCreate[%+v], error[%w]", create, err)
+	}
+	backup, err := s.composeBackup(ctx, backupRaw)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to compose Backup with backupRaw[%+v], error[%w]", backupRaw, err)
+	}
+	return backup, nil
+}
+
+// GetBackupByID gets an instance of Backup by ID
+func (s *Store) GetBackupByID(ctx context.Context, id int) (*api.Backup, error) {
+	backupRaw, err := s.getBackupRawByID(ctx, id)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get backup setting by ID[%d], error[%w]", id, err)
+	}
+	backupSetting, err := s.composeBackup(ctx, backupRaw)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to compose Backup with backupRaw[%+v], error[%w]", backupRaw, err)
+	}
+	return backupSetting, nil
+}
+
+// FindBackup finds a list of Backup instances
+func (s *Store) FindBackup(ctx context.Context, find *api.BackupFind) ([]*api.Backup, error) {
+	backupRawList, err := s.findBackupRaw(ctx, find)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to find Backup list with BackupFind[%+v], error[%w]", find, err)
+	}
+	var backupList []*api.Backup
+	for _, raw := range backupRawList {
+		backup, err := s.composeBackup(ctx, raw)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to compose Backup with backupRaw[%+v], error[%w]", raw, err)
+		}
+		backupList = append(backupList, backup)
+	}
+	return backupList, nil
+}
 
-// BackupService represents a service for managing backup.
-type BackupService struct {
-	l     *zap.Logger
-	db    *DB
-	store *Store
+// PatchBackup patches an instance of Backup
+func (s *Store) PatchBackup(ctx context.Context, patch *api.BackupPatch) (*api.Backup, error) {
+	backupRaw, err := s.patchBackupRaw(ctx, patch)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to patch Backup with BackupPatch[%+v], error[%w]", patch, err)
+	}
+	backup, err := s.composeBackup(ctx, backupRaw)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to compose Backup with backupRaw[%+v], error[%w]", backupRaw, err)
+	}
+	return backup, nil
 }
 
-// NewBackupService returns a new instance of BackupService.
-func NewBackupService(logger *zap.Logger, db *DB, store *Store) *BackupService {
-	return &BackupService{l: logger, db: db, store: store}
+// GetBackupSettingByID gets an instance of BackupSetting by ID
+func (s *Store) GetBackupSettingByID(ctx context.Context, id int) (*api.BackupSetting, error) {
+	backupSettingRaw, err := s.getBackupSettingRaw(ctx, &api.BackupSettingFind{ID: &id})
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get backup setting by ID[%d], error[%w]", id, err)
+	}
+	backupSetting, err := s.composeBackupSetting(ctx, backupSettingRaw)
```

Comment:
this will fail when backupSettingRaw is nil. Also check other places.

Reply:
Fixed. I also checked other places. PTAL.